src/formatter.rs

```diff
@@ -109,89 +108,101 @@ fn clean_line_size(line: &str) -> usize {
+        (ostyle.bold, "<<", ">>"),
+    ];
+
+    let mut cur_style = Style::default();
+    let mut buffer = String::new();
+    let mut result = String::new();
+    let mut found = false;
+    let mut stack: Vec<(Style, &str)> = vec![]; // Stack of (pre_style, opening_tag)
+
+    // Assumes no nesting <<a[[b>>c]] - should warn if it finds one ?
+    for (i, c) in text.char_indices() {
+        let slice = &text[i..];
+        // If found is not set true by the end of the iteration, we add the char to the
+        // buffer When found is true, we do nothing the next iteration since
+        // tags are two chars long
+        if found {
+            found = false;
+            continue;
+        }
+
+        for (style, tag_open, tag_close) in &tag_pairs {
+            // There's a chance of a tag starting here
+            if slice.starts_with(tag_open) {
+                // There's definitely a tag to be parsed (grouped non-lazily)
+                if slice.contains(tag_close) {
+                    // NOTE (CG RULES):
+                    // Tags can not nest themselves:
+                    // so if the current open was already in the stack - ignore
+                    // <<<<Prompt>>> => [<<Prompt]>>
+                    // Although OUR double parsing looks better imo
+                    // https://www.codingame.com/contribute/view/70888dd5bb12f2becdad5e6db3de8b40a77f
+
+                    // Paint the previous buffer with the previous colour
+                    // add it to the global "result" and then clear it
+                    result += &cur_style.paint(&buffer).to_string();
+                    buffer.clear();
+                    // push cur_style to the stack to go back to it later on
+                    // then update the color to paint the next buffer
+                    stack.push((cur_style.clone(), tag_open));
+                    cur_style = merge_styles(&cur_style, &style);
+                    // Found a tag, 2 turns skip
+                    found = true;
+                }
+            // There's a chance of a tag ending here
+            } else if slice.starts_with(tag_close) {
+                // Does this closing tag match the top of the stack?
+                match stack.clone().last() {
+                    // We did parse some opening tag before
+                    Some((st, op)) => {
+                        // They match, pop
+                        if op == tag_open {
+                            stack.pop();
+                            // Paint and go back to the previous style
+                            result += &cur_style.paint(&buffer).to_string();
+                            buffer.clear();
+                            cur_style = st.clone();
+                            found = true;
+                        } else {
+                            // They don't match <<a]]b>> treat ]] as string
+                            panic!("Nesting different tags is UNSUPPORTED")
```

Comment:
We shouldn't ever `panic!` – it's better to show something wonky (or even completely skip formatting) than not show anything at all. I would change this to print a warning to stderr.